server/src/main/java/com/exacaster/lighter/application/sessions/SessionService.java

```diff
@@ -102,4 +114,14 @@ public Statement getStatement(String id, String statementId) {
     public Statement cancelStatement(String id, String statementId) {
         return statementHandler.cancelStatement(id, statementId);
     }
+
+    public synchronized Statement executeStatement(Statement statement) {
```

Comment:
Clients should use same API, as we have for simple sessions. Post statement, and then poll statement results.

Reply:
Ok